src/utilities/buildASTSchema.js

```diff
@@ -116,9 +117,7 @@ export function buildASTSchema(
   ast: DocumentNode,
   options?: BuildSchemaOptions,
 ): GraphQLSchema {
-  if (!ast || ast.kind !== Kind.DOCUMENT) {
-    throw new Error('Must provide a document ast.');
-  }
```

Comment:
Make it similar to this:
https://github.com/graphql/graphql-js/blob/3fdf240234445789b6b876e39f2bb9ed3a977387/src/utilities/extendSchema.js#L105-L108

Reply:
Is `invariant` better, or is `GraphQLError` with the `documentAST` better? Internally, we've found converting as many `invariant(<invariant>, msg)` calls to `if (!<invariant>) { throw new GraphQLError(msg, astNode); }` has greatly reduced our tooling support burden: people have mostly stopped asking why the infra failed when their GraphQL document is invalid.

In this case, I'm not sure whether throwing `GraphQLError` would be more or less confusing, though: the callee is passing in null or a bad AST node, so it really is a tool-builder error, not a GraphQL-user error.